File: java/src/peachtree/options/BooleanOption.java
package peachtree.options;


import org.json.JSONObject;

public class BooleanOption extends Option {
	
	
	boolean value;
	
	public BooleanOption(String name, String section, String title, boolean value) {
		this(name, section, title, value, false);
	}
	
	
	public BooleanOption(String name, String section, String title, boolean value, boolean hide) {
		this.name = name;
		this.section = section;
		this.title = title; 
		this.value = value;
		this.hide = hide;
	}
	
	
	/**
	 * Set the value
	 * @return
	 */
	public boolean getVal() {
		return this.value;
	}
	
	/**
	 * Get the value
	 * @param value
	 */
	public void setVal(boolean value) {
		this.value = value;
	}

	@Override
	public JSONObject toJSON() {
		JSONObject json = new JSONObject().put("value", value);
		super.modifyJSON(json);
		return json;
	}

}

File: cpp/peachtree/src/options/BooleanOption.cpp
/*
 * BooleanOption.cpp
 *
 *  Created on: 7/07/2021
 *      Author: jdou557
 */

#include "BooleanOption.h"

BooleanOption::BooleanOption(string name, string section, string title, bool value){
	this->name = name;
	this->section = section;
	this->title = title;
	this->value = value;
	this->hidden = false;
}


BooleanOption::BooleanOption(string name, string section, string title, bool value, bool hidden){
	this->name = name;
	this->section = section;
	this->title = title;
	this->value = value;
	this->hidden = hidden;
}


jsonObject BooleanOption::toJSON() {
	jsonObject json;
	json["value"] = this->value;
	json["type"] = "BooleanOption";
	json = modifyJSON(json);
	return json;
}



bool BooleanOption::getVal() {
	return this->value;
}

void BooleanOption::setVal(bool value) {
	this->value = value;
}

